### DISCUSSION TITLE
nuclei-templates are not installing

### DISCUSSION THREAD

**Discussion by alagujeeva** [projectdiscovery/nuclei#4854] [Q&A]
Hello All - I am facing the same issue as described here - https://github.com/projectdiscovery/nuclei/discussions/3636?sort=top.

Error log:

```
[INF] nuclei-templates are not installed, installing...
[INF] Your current nuclei-templates  are outdated. Latest is v9.7.7
[ERR] Could not find template '/home/ubuntu/nuclei-templates': no templates found in path /home/ubuntu/nuclei-templates
[INF] Current nuclei version: v3.1.10 (latest)
[INF] Current nuclei-templates version:  (outdated)
[WRN] Scan results upload to cloud is disabled.
[INF] No results found. Better luck next time!
[FTL] Could not run nuclei: no templates provided for scan
```

Setup: Ubuntu 22.04
Even if I run the command "nuclei -ut" templates are not installing.

Tried the same with multiple versions of golang 1.21.5, 1.21.6, 1.22.1.

Can someone please assist?

**Comment by olearycrew** [comment]
Hi @alagujeeva 

Could you run the update template command again with `-v` like this:

```
nuclei -ut -v
```

Then if you could post the output of that here, I'm hoping maybe we see something that can help us troubleshoot more.

**Comment by olearycrew** [comment]
Also if you could provide the output of a healthcheck (`nuclei -hc`) that would help too!

**Reply by alagujeeva** [reply]
Hello - Thank you for your quick response!

nuclei -hc results

```
Version: v3.1.10
Operating System: linux
Architecture: amd64
Go Version: go1.21.5
Compiler: gc
File "/home/ubuntu/.config/nuclei/config.yaml" Read => Ok
File "/home/ubuntu/.config/nuclei/config.yaml" Write => Ok
File "/home/ubuntu/.config/nuclei/.nuclei-ignore" Read => Ok
File "/home/ubuntu/.config/nuclei/.nuclei-ignore" Write => Ok
File "/home/ubuntu/nuclei-templates/.checksum" Read => Ko (open /home/ubuntu/nuclei-templates/.checksum: no such file or directory)
File "/home/ubuntu/nuclei-templates/.checksum" Write => Ko (open /home/ubuntu/nuclei-templates/.checksum: no such file or directory)
IPv4 connectivity to scanme.sh:80 => Ok
IPv6 connectivity to scanme.sh:80 => Ko (dial tcp6 [2400:6180:0:d0::91:1001]:80: connect: network is unreachable)
IPv4 UDP connectivity to scanme.sh:53 => Ok
```

nuclei -ut -v (Tried this with sudo as well, received the same error)

```
[INF] nuclei-templates are not installed, installing...
[WRN] failed to install nuclei templates: [:RUNTIME] failed to install templates at /home/ubuntu/nuclei-templates <- [:RUNTIME] failed to write templates to disk at /home/ubuntu/nuclei-templates <- [:RUNTIME] failed to download templates <- [:RUNTIME] failed to read resp body <- context deadline exceeded (Client.Timeout or context cancellation while reading body)
[INF] Your current nuclei-templates  are outdated. Latest is v9.7.7
[WRN] failed to update nuclei templates: [:RUNTIME] failed to download templates <- [:RUNTIME] failed to read resp body <- context deadline exceeded (Client.Timeout or context cancellation while reading body)
```

Pls assist with next steps and commands. Thanks!

**Reply by olearycrew** [reply]
Interesting that there is no `.checksum` file in the nuclei templates repo.

Can you try this: 
![CleanShot 2024-03-08 at 14 00 03](https://github.com/projectdiscovery/nuclei/assets/6044920/c084f744-d354-45e6-8fda-b1d623ae28f2)

```
rm -rf ~/nuclei-templates
nuclei -ut -v
```

Also do you have a `GITHUB_TOKEN` set in your environment? If so and it is invalid that may be causing an error. So also try this:

```
GITHUB_TOKEN= nuclei -ut
```

**Reply by alagujeeva** [reply]
I am able to successfully install nuclei-templates. The issue is, it's a new VM Which do not have my git configured. 
Thank you very much!

**Reply by jfarl** [reply]
> I am able to successfully install nuclei-templates. The issue is, it's a new VM Which do not have my git configured. Thank you very much!

I'm having the same issue. What did you configure?
